system-linux.c

```diff
@@ -1691,9 +1691,87 @@ int system_vlandev_del(struct device *vlandev)
 	return system_link_del(vlandev->ifname);
 }
 
+static inline int ethtool_link_mode_test_bit(unsigned int nr, const __u32 *mask)
+{
+	return !!(mask[nr / 32] & (1 << (nr % 32)));
+}
+
+static inline int ethtool_link_mode_set_bit(unsigned int nr, __u32 *mask)
+{
+	mask[nr / 32] |= (1 << (nr % 32));
+	return 0;
+}
+
+static int
+system_set_ethtool_settings_future(struct device *dev, struct device_settings *s)
```

Comment:
Would there be some better name for this function? I'm a bit confused by the `_future` part. I don't have any better suggestions though.

Reply:
I see two options here:

* either rename current `system_set_ethtool_settings()` to `system_set_ethtool_settings_legacy()`
* rename it to `system_set_ethtool_ksettings()` or `system_set_ethtool_using_ksettings()` to model the name after [the new kernel API](https://github.com/torvalds/linux/commit/3f1ac7a700d039c61d8d8b99f28d605d489a60cf).